### PR TITLE
Replace ArrayList and LinkedList usage with GlueList

### PR DESCRIPTION
**Pull request by MrIvanPlays**
Only for our patches, if we find any hot code that needs such replacement we will do it

### CODE REVIEW

File: patches/server/0035-lithium-collision-optimizations.patch [hunk 1]
```diff
@@ -925,7 +926,7 @@ index 1250c3cbe915815939627701c153ba6254fc05f0..e0cbcc0a15f8089c29957badc20b0a78
 -                return (List) stream.filter(oclass1::isInstance).collect(Collectors.toList());
 +            List<S> list = (List<S>) this.a.computeIfAbsent(oclass, (oclass1) -> { // Yatopia - decompile fix
 +                // Yatopia start - how about we nuke stream?
-+                List<T> ret = Lists.newArrayList();
++                List<T> ret = new net.yatopia.server.list.GlueList<>();
```
Comment (foss-mc) [inline patches/server/0035-lithium-collision-optimizations.patch:929]:
```suggestion
+                net.yatopia.server.list.GlueList<T> ret = new net.yatopia.server.list.GlueList<>();
```

File: patches/server/0038-Fix-recipe-crash.patch [hunk 2]
```diff
@@ -32,7 +32,7 @@ index 64adf87f2fe9921e49bf7a76170bb9908dfdaf19..a707c85f2f6b0177781755decc8bc2e0
 +        // Yatopia start - WHY?! WHO DID THIS TO YOU!?!
 +        Collection<IRecipe<C>> recipeCollection = this.b(recipes).values();
 +        if (recipeCollection.isEmpty()) return java.util.Collections.emptyList();
-+        List<T> ret = new java.util.ArrayList<>();
++        List<T> ret = new net.yatopia.server.list.GlueList<>();
```
Comment (foss-mc) [inline patches/server/0038-Fix-recipe-crash.patch:35]:
```suggestion
+        net.yatopia.server.list.GlueList<T> ret = new net.yatopia.server.list.GlueList<>();
```

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch [hunk 3]
```diff
@@ -38,7 +38,7 @@ index 6fcc7ed7c129e6a33386d65b37cbba4a44e96f0f..e6b5a21c523c598f53207d024322301f
  
 +    // Yatopia start
 +    public static java.util.List<BlockPosition> getPositions(int i, int j, int k, int l, int i1, int j1) {
-+        java.util.List<BlockPosition> ret = new java.util.ArrayList<>();
++        java.util.List<BlockPosition> ret = new net.yatopia.server.list.GlueList<>();
```
Comment (foss-mc) [inline patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:41]:
```suggestion
+        net.yatopia.server.list.GlueList<BlockPosition> ret = new net.yatopia.server.list.GlueList<>();
```

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch [hunk 4]
```diff
@@ -83,7 +83,7 @@ index b1ff30fb1569ddf59f46240266ce32f0aa96da1a..68deb623508326841cf750fca45b5384
              }, blockposition, 20, VillagePlace.Occupancy.ANY);
 -
 +            // Yatopia start
-+            List<BlockPosition> ret = new java.util.ArrayList<>();
++            List<BlockPosition> ret = new net.yatopia.server.list.GlueList<>();
```
Comment (foss-mc) [inline patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:86]:
```suggestion
+            net.yatopia.server.list.GlueList<BlockPosition> ret = new net.yatopia.server.list.GlueList<>();
```

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch [hunk 5]
```diff
@@ -192,7 +192,7 @@ index 6ab80cf7908437309a7346d71e42cf128a6f13d9..07a36d1955037524715181461cc2aa5d
 +        int j = Math.floorDiv(i, 16) + 1;
 +
 +        java.util.List<ChunkCoordIntPair> list = ChunkCoordIntPair.streamList(new ChunkCoordIntPair(pos), j);
-+        java.util.List<VillagePlaceRecord> ret = new java.util.ArrayList<>();
++        java.util.List<VillagePlaceRecord> ret = new net.yatopia.server.list.GlueList<>();
```
Comment (foss-mc) [inline patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:195]:
```suggestion
+        net.yatopia.server.list.GlueList<VillagePlaceRecord> ret = new net.yatopia.server.list.GlueList<>();
```

File: patches/server/0035-lithium-collision-optimizations.patch (hunk 1 above)
Comment (MrIvanPlays) [reply patches/server/0035-lithium-collision-optimizations.patch:929]: no

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch [hunk 6]
```diff
@@ -202,7 +202,7 @@ index 6ab80cf7908437309a7346d71e42cf128a6f13d9..07a36d1955037524715181461cc2aa5d
 +    }
 +    public java.util.List<VillagePlaceRecord> cList(Predicate<VillagePlaceType> predicate, BlockPosition pos, int i, VillagePlace.Occupancy occupancy) {
 +        int j = i * i;
-+        java.util.List<VillagePlaceRecord> ret = new java.util.ArrayList<>();
++        java.util.List<VillagePlaceRecord> ret = new net.yatopia.server.list.GlueList<>();
```
Comment (foss-mc) [inline patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:205]:
```suggestion
+        net.yatopia.server.list.GlueList<VillagePlaceRecord> ret = new net.yatopia.server.list.GlueList<>();
```

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch [hunk 7]
```diff
@@ -218,7 +218,7 @@ index 6ab80cf7908437309a7346d71e42cf128a6f13d9..07a36d1955037524715181461cc2aa5d
  
 +    // Yatopia start
 +    public java.util.List<BlockPosition> aList(Predicate<VillagePlaceType> predicate, Predicate<BlockPosition> posFilter, BlockPosition pos, int i, VillagePlace.Occupancy occupancy) {
-+        java.util.List<BlockPosition> ret = new java.util.ArrayList<>();
++        java.util.List<BlockPosition> ret = new net.yatopia.server.list.GlueList<>();
```
Comment (foss-mc) [inline patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:221]:
```suggestion
+        net.yatopia.server.list.GlueList<BlockPosition> ret = new net.yatopia.server.list.GlueList<>();
```

File: patches/server/0038-Fix-recipe-crash.patch (hunk 2 above)
Comment (MrIvanPlays) [reply patches/server/0038-Fix-recipe-crash.patch:35]: no

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch [hunk 8]
```diff
@@ -270,7 +270,7 @@ index 6ab80cf7908437309a7346d71e42cf128a6f13d9..07a36d1955037524715181461cc2aa5d
      public Optional<BlockPosition> a(Predicate<VillagePlaceType> predicate, Predicate<BlockPosition> predicate1, BlockPosition blockposition, int i) {
 +        // Yatopia start
 +        java.util.List<VillagePlaceRecord> list = this.cList(predicate, blockposition, i, VillagePlace.Occupancy.HAS_SPACE);
-+        java.util.List<VillagePlaceRecord> filtered = new java.util.ArrayList<>();
++        java.util.List<VillagePlaceRecord> filtered = new net.yatopia.server.list.GlueList<>();
```
Comment (foss-mc) [inline patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:273]:
```suggestion
+        net.yatopia.server.list.GlueList<VillagePlaceRecord> filtered = new net.yatopia.server.list.GlueList<>();
```

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch (hunk 3 above)
Comment (MrIvanPlays) [reply patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:41]: no

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch [hunk 9]
```diff
@@ -349,7 +349,7 @@ index 77c66bc9952542d2444b402896a3d9f622ca2ff9..b47abdabb592cfff6a4bf9ee638dd72f
  
 +    // Yatopia start
 +    public java.util.List<VillagePlaceRecord> aList(Predicate<VillagePlaceType> predicate, VillagePlace.Occupancy occupancy) {
-+        java.util.List<VillagePlaceRecord> ret = new java.util.ArrayList<>();
++        java.util.List<VillagePlaceRecord> ret = new net.yatopia.server.list.GlueList<>();
```
Comment (foss-mc) [inline patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:352]:
```suggestion
+        net.yatopia.server.list.GlueList<VillagePlaceRecord> ret = new net.yatopia.server.list.GlueList<>();
```

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch (hunk 4 above)
Comment (MrIvanPlays) [reply patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:86]: no

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch (hunk 5 above)
Comment (MrIvanPlays) [reply patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:195]: no

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch (hunk 6 above)
Comment (MrIvanPlays) [reply patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:205]: no

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch (hunk 7 above)
Comment (MrIvanPlays) [reply patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:221]: no

File: patches/server/0040-Highly-optimize-VillagePlace-filtering.patch (hunk 8 above)
Comment (MrIvanPlays) [reply patches/server/0040-Highly-optimize-VillagePlace-filtering.patch:273]: no

**Pull request merged**